src/node/build/index.ts

```diff
@@ -771,12 +771,14 @@ function createEmitPlugin(
  */
 function getEmitIndexPath(
   indexPath: string,
+  root: string,
   outDir: string,
   { input, output }: Build
 ) {
-  return typeof input === 'string' && path.resolve(input) === indexPath
-    ? path.resolve(outDir, output.file || input)
```

Comment:
Here will overwrite original input file.

Reply:
Oops, good catch! 

Renaming to `index` feels unexpected. If the `entry` has a name other than `index`, shouldn't we assume the developer wants to keep it that way? 

You can set `rollupOutputOptions: { file: 'index.html' }` in Vite config to be explicit about renaming the compiled index. Maybe we can support `entry: { 'input.html': 'output.html' }` at some point.

```ts
if (typeof input === 'string') {
  // Rollup "input" option is relative to working directory.
  input = path.resolve(input)

  if (input === indexPath) {
    return path.resolve(outDir, output.file || path.relative(root, input))
  }
}
return ''
```